diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt

```diff
@@ -130,20 +133,24 @@ class IdentifierNaming : Rule("identifier-naming") {
     }
 
     /**
-     * getting identifiers (aka variable names) from parent nodes like PROPERTY
-     * The trick here is to handle DESTRUCTURING_DECLARATION correctly, as it does not have IDENTIFIER leaf
+     * Getting identifiers (aka variable names) from parent nodes like PROPERTY.
+     * Several things to take into account here:
+     *     * need to handle DESTRUCTURING_DECLARATION correctly, as it does not have IDENTIFIER leaf.
+     *     * function type can have VALUE_PARAMETERs without name
      */
     private fun extractVariableIdentifiers(node: ASTNode): List<ASTNode> {
-        val destructingDeclaration = node.getFirstChildWithType(ElementType.DESTRUCTURING_DECLARATION)
+        val destructingDeclaration = node.getFirstChildWithType(DESTRUCTURING_DECLARATION)
         val result = if (destructingDeclaration != null) {
-            destructingDeclaration.getAllChildrenWithType(ElementType.DESTRUCTURING_DECLARATION_ENTRY)
+            destructingDeclaration.getAllChildrenWithType(DESTRUCTURING_DECLARATION_ENTRY)
                     .map { it.getIdentifierName()!! }
+        } else if (node.treeParent.elementType == VALUE_PARAMETER_LIST && node.treeParent.treeParent.elementType == FUNCTION_TYPE) {
```

Comment:
I am afraid that this check on `file level` can cause troubles. What if node (file level constant) will not have treeParent.treeParent?

Reply:
Not sure, if such case can actually happen, but I have still added a check for this.

Reply:
so?

Reply:
https://github.com/cqfn/diKTat/pull/117/commits/1d24f332024ed8f42f4233b1f3d7fb3be2c21ee0